Tests/DatadogCrashReportingTests/DDCrashReportingPluginTests.swift

```diff
@@ -0,0 +1,140 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+* This product includes software developed at Datadog (https://www.datadoghq.com/).
+* Copyright 2019-2020 Datadog, Inc.
+*/
+
+import XCTest
+@testable import DatadogCrashReporting
+
+class DDCrashReportingPluginTests: XCTestCase {
+    override func setUp() {
+        super.setUp()
+        XCTAssertNil(DDCrashReportingPlugin.thirdPartyCrashReporter)
+    }
+
+    override func tearDown() {
+        XCTAssertNil(DDCrashReportingPlugin.thirdPartyCrashReporter)
+        super.tearDown()
+    }
+
+    // MARK: - Processing Crash Report in Caller
+
+    func testGivenPendingCrashReport_whenCallerSucceedsWithItsProcessing_itIsPurged() throws {
+        let expectation = self.expectation(description: "Crash Report was delivered to the caller.")
+        let crashReporter = try ThirdPartyCrashReporterMock()
+        let plugin = DDCrashReportingPlugin { crashReporter }
+        defer { DDCrashReportingPlugin.thirdPartyCrashReporter = nil }
+
+        // Given
+        crashReporter.pendingCrashReport = .mockAny()
+
+        // When
+        plugin.readPendingCrashReport { crashReport in
+            XCTAssertEqual(crashReport, crashReporter.pendingCrashReport)
+            expectation.fulfill()
+            return true // the caller succeeded in processing the crash report
+        }
+
+        // Then
+        waitForExpectations(timeout: 0.5, handler: nil)
```

Comment:
I see that this should automatically work once `readPendingCrashReport()` becomes async 👍